stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -872,6 +872,39 @@ function dot(x::AbstractArray, y::AbstractArray)
     s
 end
 
+"""
+    dot(x, A, y)
+
+Compute the generalized dot product `dot(x, A*y)` between two vectors `x` and `y`,
+without storing the intermediate result of `A*y`. As for the two-argument
+[`dot(_,_)`](@ref), this acts recursively. Moreover, for complex vectors, the
+first vector is conjugated.
+"""
+function dot end
```

Comment:
Actually, probably better to attach this to the generic fallback `dot(x, A, y) = dot(x, A*y)` definition, so that it doesn't conflict with the 2-argument `dot` docstring.